File: java/src/main/java/vsdk/toolkit/io/context/TransformSequenceContext.java
package vsdk.toolkit.io.context;

public class TransformSequenceContext {
    public static final int TRANSFORM_MAXIMUM_DIMENSIONS = 8;

    public FilePositionContext startingPosition; // Starting position on input
    public int numberOfDimensions; // Number of array dimensions
    public TransformArrayContext[] transformArguments;

    public TransformSequenceContext() {
        startingPosition = new FilePositionContext();
        numberOfDimensions = 0;
        transformArguments = new TransformArrayContext[TRANSFORM_MAXIMUM_DIMENSIONS];
        for (int i = 0; i < transformArguments.length; i++) {
            transformArguments[i] = new TransformArrayContext();
        }
    }
}

File: typescript/base/src/main/vsdk/toolkit/io/context/TransformSequenceContext.ts
import { FilePositionContext } from "./FilePositionContext";
import { TransformArrayContext } from "./TransformArrayContext";

export class TransformSequenceContext {
  public static readonly TRANSFORM_MAXIMUM_DIMENSIONS = 8;

  public startingPosition: FilePositionContext;
  public numberOfDimensions: number;
  public transformArguments: TransformArrayContext[];

  public constructor() {
    this.startingPosition = new FilePositionContext();
    this.numberOfDimensions = 0;
    this.transformArguments = new Array<TransformArrayContext>(TransformSequenceContext.TRANSFORM_MAXIMUM_DIMENSIONS);
    for (let i = 0; i < this.transformArguments.length; i++) {
      this.transformArguments[i] = new TransformArrayContext();
    }
  }
}
